### PR TITLE
Fix Issue #80 inserting every fontUsage (including unused font) for multiple html target

### PR DESCRIPTION
**Pull request by palindrom615**

### CODE REVIEW

File: test/subsetFonts.js [hunk 1]
```diff
@@ -4250,9 +4250,9 @@ describe('subsetFonts', function() {
       });
     });
 
-    describe('with a page that does need subsetting and one that does', function() {
+    describe('with a page that does NOT need subsetting and one that does', function() {
       // https://gitter.im/assetgraph/assetgraph?at=5dbb6438a3f0b17849c488cf
-      it('should not short circuit because the first page does not need any subset fonts', async function() {
+      it('should not short circuit with explicit subsetPerPage:true option because the first page does not need any subset fonts', async function() {
```
Comment (papandreou) [inline test/subsetFonts.js:4255]: This probably deserves to be its own test case alongside the existing one.

However, if I copy your test over to master, it also passes? Can you come up with an actual regression test?

File: lib/subsetFonts.js [hunk 2]
```diff
@@ -728,26 +728,29 @@ async function subsetFonts(
...
-    // Gather all texts
-    for (const htmlAssetTextWithProps of htmlAssetTextsWithProps) {
-      for (const fontUsage of htmlAssetTextWithProps.fontUsages) {
-        if (!globalFontUsage[fontUsage.fontUrl]) {
-          globalFontUsage[fontUsage.fontUrl] = [];
+    const fontUsagesConcat = _.flatMap(
+      htmlAssetTextsWithProps,
+      htmlAssetTextWithProps => htmlAssetTextWithProps.fontUsages
+    );
+    const globalFontUsageMap = fontUsagesConcat.reduce(
```
Comment (papandreou) [inline lib/subsetFonts.js:735]: Mind turning this `reduce` construct into a regular `for`...`of` loop? I find those much easier to reason about.

File: lib/subsetFonts.js [hunk 3]
```diff
@@ -728,26 +728,29 @@ async function subsetFonts(
...
+      (fontUsageMap, fontUsage) => {
+        const newFontUsage = fontUsage;
+        if (fontUsage.fontUrl in fontUsageMap) {
+          newFontUsage.text = _.uniq(
+            fontUsageMap[fontUsage.fontUrl].text + fontUsage.text
+          )
+            .sort()
+            .join('');
```
Comment (papandreou) [inline lib/subsetFonts.js:743]: Hmm, mental note that we should find a better way achieve the "unique across pages" thing, sorting the output of `_.uniq` breaks when there's multiple non-UTF-16 characters:

```js
require('lodash').uniq('🤗').sort().join('')
'🤗'
require('lodash').uniq('🤗🤞').sort().join('')
'🤗�'
require('lodash').uniq('🤗🤞👊').sort().join('')
'�🡊��'
```

Comment (Munter) [reply lib/subsetFonts.js:743]: This doesn't have the same problem. Not very readable though:
```js
[...new Set([...'🤗🤞👊'])].sort().join('')
```

Comment (papandreou) [reply lib/subsetFonts.js:743]: Nice! I think we should go with that, then. Not readable is better than broken :)

Comment (papandreou) [reply lib/subsetFonts.js:743]: Implemented your fix here: https://github.com/Munter/subfont/pull/87

**Comment by papandreou** [comment]
Replaced by https://github.com/Munter/subfont/pull/111

**Pull request closed without merging**